File: backend/src/ZealEducation.Domain/Entities/Examination.cs
using ZealEducation.Domain.Common;

namespace ZealEducation.Domain.Entities;

public class Examination : BaseAuditableEntity
{
    public Guid BatchId { get; set; }
    public string ExamName { get; set; } = default!;
    public DateOnly ExamDate { get; set; }
    public string? Location { get; set; }
    public int MaxScore { get; set; } = 100;
    public int PassScore { get; set; } = 50;
    public Guid ScheduledById { get; set; }

    public Batch Batch { get; set; } = default!;
    public Staff ScheduledBy { get; set; } = default!;
    public ICollection<ExamResult> ExamResults { get; set; } = [];
}

File: frontend/zeal-edu/src/app/core/models/examination.ts
export interface Examination {
  examinationId: string;
  batchId: string;
  examName: string;
  examDate: string;
  location: string | null;
  maxScore: number;
  passScore: number;
  scheduledById: string;
  scheduledByName: string;
  hasResults: boolean;
  createdAt: string;
}

export interface CreateExaminationPayload {
  examName: string;
  examDate: string;
  location: string | null;
  maxScore: number;
  passScore: number;
}

export interface UpdateExaminationPayload extends CreateExaminationPayload {}
